src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -759,21 +768,146 @@ public boolean isKnownDatasource( UUID studyId, String participantId, String dat
                 && participantDevices.get( participantId ).containsKey( datasourceId );
     }
 
+    private void deleteStudyData(
+            UUID studyId,
+            java.util.Optional<String> participantId,
+            com.openlattice.data.DeleteType deleteType,
+            String userToken ) {
+        try {
+            // load api for actions authenticated by the user
+            ApiClient userApiClient = new ApiClient( () -> userToken
+            );
```

Comment:
Another tiny style thing, but can you move this close parenthesis to the previous line?